platform/golden/bullseye.pkr.hcl

```diff
@@ -76,37 +90,49 @@ source "virtualbox-iso" "base-debian-amd64" {
   skip_nat_mapping     = false
   iso_checksum         = "${var.iso_checksum_type}:${var.iso_checksum}"
   iso_url              = "${var.iso_url}"
-  iso_interface        = "virtio"
+  iso_interface        = "sata"
   nic_type             = "virtio"
   hard_drive_interface = "virtio"
   rtc_time_base        = "UTC"
   shutdown_command     = "sudo -S shutdown -P now"
   ssh_username         = "packer"
   ssh_password         = "${var.ssh_password}"
   ssh_wait_timeout     = "30m"
+  firmware             = "efi"
+  # don't remove the built VM from VirtualBox after export
+  keep_registered      = "false"
+  # don't export the built VM
+  skip_export          = "false"
   vboxmanage = [
     # enable recording video of install process, for debug and build record
     [ "modifyvm", "{{.Name}}", "--recording", "on" ],
-    # enable proxy access across NAT network interface.
     [ "modifyvm", "{{.Name}}", "--nat-localhostreachable1", "on" ],
+    [ "modifyvm", "{{.Name}}", "--firmware", "EFI" ],
```

Comment:
I am 99% positive that line 101 ensures the vm is created with EFI, and but this line at 110 may be modifying the boot type after the image is created. I have read the code, and this part seems broken, but I am not 100% certain. Can you try without. line 110?

The warning from the documentation:
```
o pcie, the drive is attached to an NVMe controller. When set to virtio, the drive is attached to a VirtIO controller. Please note that when you use "pcie", you'll need to have Virtualbox 6, install an [extension pack](https://www.virtualbox.org/wiki/Downloads#VirtualBox6.0.14OracleVMVirtualBoxExtensionPack) and you will need to enable EFI mode for nvme to work, ex:
in HCL:.
          [ "modifyvm", "{{.Name}}", "--firmware", "EFI" ],
```

I have read that warning hundreds of times, and every time it is inconsistent, and I finally just during this review determined why.

This is a virtualbox **SIX** problem. We use vbox7.

Reply:
I'll see about it shortly and leave another comment here.

Reply:
Yep, after adding the firmware = "efi" boot_command pamareter, it is no longer necessary to set the EFI firmware via modifyvm.